x-pack/test/functional/services/aiops/explain_log_rate_spikes.ts

```diff
@@ -56,9 +58,93 @@ export function ExplainLogRateSpikesProvider({ getService }: FtrProviderContext)
+        `aiopsRerunAnalysisButton${shouldRerun ? ' shouldRerun' : ''}`
+      );
+
+      await retry.tryForTime(30 * 1000, async () => {
+        await testSubjects.existOrFail(
+          `aiopsRerunAnalysisButton${!shouldRerun ? ' shouldRerun' : ''}`
+        );
+      });
+    },
+
+    async assertAnalysisSectionExist() {
+      await retry.tryForTime(5000, async () => {
+        await testSubjects.existOrFail(`aiopsExplainLogRateSpikesAnalysis`);
+      });
+    },
+
+    async assertRerunAnalysisButton(shouldRerun: boolean) {
+      await testSubjects.existOrFail(
+        `aiopsRerunAnalysisButton${shouldRerun ? ' shouldRerun' : ''}`
+      );
+    },
+
+    async assertProgressTitle(expectedProgressTitle: string) {
+      await testSubjects.existOrFail('aiopProgressTitle');
+      const currentProgressTitle = await testSubjects.getVisibleText('aiopProgressTitle');
+      expect(currentProgressTitle).to.be(expectedProgressTitle);
+    },
+
     async navigateToIndexPatternSelection() {
       await testSubjects.click('mlMainTab explainLogRateSpikes');
       await testSubjects.existOrFail('mlPageSourceSelection');
     },
+
+    async getBrushSelectionWidth(selector: string) {
+      const brush = await testSubjects.find(selector);
+      const brushSelectionBefore = (await brush.findAllByClassName('selection'))[0];
+      const brushSelectionRectBefore = await brushSelectionBefore._webElement.getRect();
+      return brushSelectionRectBefore.width;
+    },
+
+    async getPxForTimestamp(timestamp: number) {
+      await elasticChart.waitForRenderComplete('aiopsDocumentCountChart');
+      const chartDebugData = await elasticChart.getChartDebugData('aiopsDocumentCountChart');
+
+      const dualBrushWrapper = await testSubjects.find('aiopsDualBrush');
+      const dualBrushWrapperRect = await dualBrushWrapper._webElement.getRect();
+
+      const bars = chartDebugData?.bars?.[0].bars ?? [];
+      const barsCount = bars.length;
+      const targetDeviationBarIndex = bars.findIndex((b) => b.x === timestamp);
+
+      const targetPx = Math.round(
+        (targetDeviationBarIndex / barsCount) * dualBrushWrapperRect.width
+      );
+      const intervalPx = Math.round((1 / barsCount) * dualBrushWrapperRect.width);
+
+      return { targetPx, intervalPx };
+    },
+
+    async adjustBrushHandler(selector: string, handlerClassName: string, targetPx: number) {
+      const brush = await testSubjects.find(selector);
+      const dualBrushWrapper = await testSubjects.find('aiopsDualBrush');
+      const dualBrushWrapperRect = await dualBrushWrapper._webElement.getRect();
+
+      const handle = (await brush.findAllByClassName(handlerClassName))[0];
+      const handleRect = await handle._webElement.getRect();
+      const handlePx = handleRect.x - dualBrushWrapperRect.x;
+      const handleFactor = handlePx < targetPx ? 1 : -1;
+      const dragAndDropOffsetPx = Math.abs(handlePx - targetPx) * handleFactor;
```

Comment:
Couldn't these two lines of code be simplified as
```suggestion
      const dragAndDropOffsetPx = targetPx - handlePx;
```
or am I missing anything here?

Reply:
You're right it can be simplified like that! I had it like this originally but it didn't work for me, probably one of the earlier calculations for `handlePx` was wrong on top of that which works now! Simplified in [7e45cdd](https://github.com/elastic/kibana/pull/138661/commits/7e45cddfc7117a4738d37b6227e505a89a27ef5c).